storybook/qmlTests/tests/tst_ShellContainer.qml

```diff
@@ -0,0 +1,394 @@
+                console.info("!!! ITEM", key, "PINNED:", pin)
+            }
+            onDappDisconnectRequested: function(dappUrl) {
+                console.info("!!! DAPP DISCONNECT:", dappUrl)
+            }
+
+            onNotificationButtonClicked: {
+                console.info("!!! onNotificationButtonClicked")
+            }
+            onSetCurrentUserStatusRequested: function (status) {
+                profileStore.currentUserStatus = status
+                console.info("!!! onSetCurrentUserStatusRequested:", status)
+            }
+            onViewProfileRequested: function(pubKey) {
+                console.info("!!! onViewProfileRequested:", pubKey)
+            }
+        }
+    }
+
+    SignalSpy {
+        id: dynamicSpy
+
+        function setup(t, s) {
+            clear()
+            target = t
+            signalName = s
+        }
+
+        function cleanup() {
+            target = null
+            signalName = ""
+            clear()
+        }
+    }
+
+    property ShellContainer controlUnderTest: null
+
+    TestCase {
+        name: "ShellContainer"
+        when: windowShown
+
+        function init() {
+            controlUnderTest = createTemporaryObject(componentUnderTest, root)
+        }
+
+        function cleanup() {
+            dynamicSpy.cleanup()
+            shellAdaptor.clear() // cleanup the pinned items
+        }
+
+        function keyClickSequence(keys) {
+            for (let k of keys) {
+                keyClick(k)
+            }
+        }
+
+        function triggerRightClick(item, delay = 100, modifiers = Qt.NoModifier) {
+            mouseClick(item, item.width/2, item.height/2, Qt.RightButton, modifiers, delay)
+        }
+
+        function triggerLongPress(item, modifiers = Qt.NoModifier) {
+            mousePress(item, item.width/2, item.height/2, Qt.LeftButton, modifiers, Qt.styleHints.mousePressAndHoldInterval)
+            mouseRelease(item, item.width/2, item.height/2, Qt.LeftButton, modifiers, Qt.styleHints.mousePressAndHoldInterval*2)
+        }
+
+        function test_basic_geometry() {
+            verify(!!controlUnderTest)
+            verify(controlUnderTest.width > 0)
+            verify(controlUnderTest.height > 0)
+        }
+
+        function test_activityCenterButton() {
+            const btn = findChild(controlUnderTest, "shellACButton")
+            verify(!!btn)
+            dynamicSpy.setup(controlUnderTest, "notificationButtonClicked")
+            mouseClick(btn)
+            tryCompare(dynamicSpy, "count", 1)
+        }
+
+        function test_shellProfileButton() {
+            const btn = findChild(controlUnderTest, "shellProfileButton")
+            verify(!!btn)
+            mouseClick(btn)
+            const popupMenu = findChild(btn, "userStatusContextMenu")
+            verify(!!popupMenu)
+            tryCompare(popupMenu, "opened", true)
+        }
+
+        function test_gridItem_search_and_click_data() {
```

Comment:
I'm not familiar with qml tests. Does this test have some magic parsing that executes it? Because it doesn't seem to do much

edit: ah I see that the test below has the same name and a `data` param, so I assume this function's return is passed to the real test below. That's interesting

Reply:
yeah there are some magic patterns in QML tests:
- a test starts with `function test_`
- a benchmark starts with `function benchmark_`
- a data driven test function gets its `data` as a param, from the same function name plus the `_data()` appended (https://doc.qt.io/archives/qt-5.15/qml-qttest-testcase.html#data-driven-tests)